R/bundle.R

```diff
@@ -35,11 +42,18 @@ bundleApp <- function(appName, appDir, appFiles, appPrimaryDoc, assetTypeName,
                       contentCategory, accountInfo) {
 
   # infer the mode of the application from its layout
-  appMode <- inferAppMode(appDir, appFiles)
+  appMode <- inferAppMode(appDir, appPrimaryDoc, appFiles)
   hasParameters <- appHasParameters(appDir, appFiles)
 
   # copy files to bundle dir to stage
-  bundleDir <- bundleAppDir(appDir, appFiles)
+  bundleDir <- bundleAppDir(appDir, appFiles, appPrimaryDoc)
+
+  if (!isShinyapps(accountInfo)) {
+    # infer package dependencies for non-static content deployment
+    if (appMode != "static") {
+      addPackratSnapshot(bundleDir, inferDependencies(appMode, hasParameters))
```

Comment:
We are already doing `addPackratSnapshot` under `snapshotDependencies` under `createAppManifest` (and not just for non-shinyapps). Why again here?